node/bft/src/sync/mod.rs

```diff
@@ -392,48 +405,45 @@ impl<N: Network> Sync<N> {
+            if block.authority().is_quorum() {
                 // Reconstruct the unconfirmed transactions.
                 let unconfirmed_transactions = cfg_iter!(block.transactions())
                     .filter_map(|tx| {
                         tx.to_unconfirmed_transaction().map(|unconfirmed| (unconfirmed.id(), unconfirmed)).ok()
                     })
                     .collect::<HashMap<_, _>>();
-
-                // Iterate over the certificates.
-                for certificates in subdag.values().cloned() {
-                    cfg_into_iter!(certificates).for_each(|certificate| {
-                        self.storage.sync_certificate_with_block(block, certificate, &unconfirmed_transactions);
-                    });
-                }
+                // Obtain the subdag.
+                let subdag = block.to_full_subdag()?;
+                // Retrieve the leader certificate and election certificate IDs.
+                let leader_certificate = subdag.leader_batch_certificate()?.clone();
+                leader_certificates.push(leader_certificate);
 
                 // Update the validator telemetry.
                 #[cfg(feature = "telemetry")]
-                self.gateway.validator_telemetry().insert_subdag(subdag);
+                self.gateway.validator_telemetry().insert_subdag(&subdag);
+
+                // Construct iter over the certificates.
+                let certificate_iter = subdag.into_iter_batch_certificates()?;
+
+                // Iterate over the certificates.
+                for certificate in certificate_iter {
+                    self.storage.sync_certificate_with_block(block, certificate.clone(), &unconfirmed_transactions)?;
+                    // Store the certificates in the list.
+                    certificates.push(certificate);
+                }
             }
         }
 
         /* Sync the BFT DAG */
 
-        // Construct a list of the certificates.
-        let certificates = blocks
-            .iter()
-            .flat_map(|block| {
-                match block.authority() {
-                    // If the block authority is a beacon, then skip the block.
-                    Authority::Beacon(_) => None,
-                    // If the block authority is a subdag, then retrieve the certificates.
-                    Authority::Quorum(subdag) => Some(subdag.values().flatten().cloned().collect::<Vec<_>>()),
-                }
-            })
-            .flatten()
-            .collect::<Vec<_>>();
-
         // If a BFT sender was provided, send the certificates to the BFT.
         if let Some(bft_sender) = self.bft_sender.get() {
-            // Await the callback to continue.
-            if let Err(e) = bft_sender.tx_sync_bft_dag_at_bootup.send(certificates).await {
-                bail!("Failed to update the BFT DAG from sync: {e}");
+            // We need certificates to send.
+            if !certificates.is_empty() && !leader_certificates.is_empty() {
```

Comment:
Hmm this might be outdated code, I guess Kai removed this in a recent iteration, shall we remove Kai? Then I guess we also won't have to collect the `leader_certificates` above anymore.